table/tables/tables.go

```diff
@@ -233,19 +233,10 @@ func (t *Table) UpdateRecord(ctx context.Context, h int64, oldData, newData []ty
 	for _, col := range t.WritableCols() {
 		var value types.Datum
 		if col.State != model.StatePublic {
-			// If col is in write only or write reorganization state
-			// and the value is not default, keep the original value.
-			value, err = table.GetColOriginDefaultValue(ctx, col.ToInfo())
-			if err != nil {
-				return errors.Trace(err)
-			}
-			cmp, errCmp := oldData[col.Offset].CompareDatum(ctx.GetSessionVars().StmtCtx, &value)
-			if errCmp != nil {
-				return errors.Trace(errCmp)
-			}
-			if cmp != 0 {
-				value = oldData[col.Offset]
-			}
+			// If col is in write only or write reorganization state we should keep the oldData.
+			// Because the oldData must be the orignal data(it's changed by other TiDBs.) or the orignal default value.
+			// TODO: Use newData directly.
+			value = oldData[col.Offset]
```

Comment:
Could we  move this update before `rebuildIndices`, although an indexed col would always be public?

Reply:
We will remove this code next time, so we should do it now.